orca-front50/src/main/groovy/com/netflix/spinnaker/orca/front50/DependentPipelineStarter.groovy

```diff
@@ -105,9 +122,24 @@ class DependentPipelineStarter implements ApplicationContextAware {
     }
 
     log.info('executing dependent pipeline {}', pipeline.id)
+    return pipeline
+  }
 
-    pipeline
+  // There are currently two sources-of-truth for the user:
+  // 1. The MDC context, which are the values that get propagated to downstream services like Front50.
+  // 2. The Execution.AuthenticationDetails object.
```

Comment:
Right, but in terms of "source of truth" to downstream services, it doesn't matter what's in the Execution.AuthenticationDetails if MDC holds different values. 

In the case of the implicit pipeline invocation, the MDC is empty, which is why we fall back to Execution.AuthenticationDetails.